File: permissions/src/main/java/com/gabriel/permissions/domain/repository/RoleRepository.java
package com.gabriel.permissions.domain.repository;

import com.gabriel.permissions.domain.model.Role;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface RoleRepository extends JpaRepository<Role, UUID> {

    Optional<Role> findByName(String name);

    void deleteByName(String name);

    void deleteAll();
}

File: permissions/src/main/kotlin/com/gabriel/permissions/domain/repository/RoleRepository.kt
package com.gabriel.permissions.domain.repository

import com.gabriel.permissions.domain.model.Role
import io.quarkus.hibernate.orm.panache.PanacheRepository
import jakarta.enterprise.context.ApplicationScoped
import java.util.UUID

@ApplicationScoped
class RoleRepository : PanacheRepository<Role> {

    fun findById(uuid: UUID): Role? = find("id", uuid).firstResult()

    fun findByName(name: String): Role? = find("name", name).firstResult()

    fun deleteByName(name: String) {
        delete("name", name)
    }

    fun deleteById(uuid: UUID) {
        delete("id", uuid)
    }
}
